### PR TITLE
(PUP-9573) Bump win32-service gem to 2.1.4

### PR DESCRIPTION
**Pull request by gimmyxd**
This also add a dependency to ffi-win32-extensions gem
which is needed by win32-service > 1.0.0

### CODE REVIEW

**Comment by gimmyxd** [comment]
Jenkins please test this on redhat7-64a,ubuntu1604-64a,windows10ent-64a,windows10ent-32a,windows2008r2-64a,windows2012r2-64a,windows2016-64a,windows2019-64a

**Comment by donoghuc** [comment]
Looks like we will need to add the ffi-win32-extension dependency to bolt-runtime as well.

**Comment by gimmyxd** [comment]
Jenkins please test this on redhat7-64a,ubuntu1604-64a,windows10ent-64a,windows10ent-32a,windows2008r2-64a,windows2012r2-64a,windows2016-64a,windows2019-64a with agent 5.5.x

**Comment by gimmyxd** [comment]
Jenkins please test this on redhat7-64a,ubuntu1604-64a,windows10ent-64a,windows10ent-32a,windows2008r2-64a,windows2012r2-64a,windows2016-64a,windows2019-64a

**Comment by gimmyxd** [comment]
@donoghuc i added also `ffi-win32-extension` dependency to `bolt-runtime`.
I'll let the split of the specific gems for windows to the bolt tea.

**Comment by gimmyxd** [comment]
runtime builds: https://jenkins-master-prod-1.delivery.puppetlabs.net/job/platform_puppet-runtime-pr-testing_runtime-pr-packaging-job/65/

runtime packages: http://builds.delivery.puppetlabs.net/puppet-runtime/4637dcdbf25b16ab97728be4d03873247fd83903/artifacts/?C=M&O=D

File: configs/projects/bolt-runtime.rb [hunk 1]
```diff
@@ -178,7 +179,7 @@
   proj.component 'rubygem-terminal-table'
   proj.component 'rubygem-unicode-display_width'
 
-  # Core Windows dependencies
+  # Core Windows dependenciesq
```
Comment (joshcooper) [inline configs/projects/bolt-runtime.rb]: nit typo

Comment (donoghuc) [reply configs/projects/bolt-runtime.rb]: Yeah it looks like the testing linked above was done against https://github.com/puppetlabs/puppet-runtime/commit/4637dcdbf25b16ab97728be4d03873247fd83903 , then PR author realized that that commit duplicated listing ffi as a dependency in bolt-rutnime (where the trivial typo was introduced). I'm currently building a new batch of bolt-runtime to validate against bolt-vanagon.

**Comment by joshcooper** [comment]
One general concern about the djberger's win32 gems is they often have problems dealing with non US-ASCII strings, like service names and descriptions. At a minimum, it would be good to verify puppet can manage services on non en_US Windows, where the service names and descriptions are localized. It also looks like more recent versions of win32-service require ruby 2.2, which could break us if puppet-agent#5.5.x relies on this version of puppet-runtime? I don't think that's the case, but wanted to call that out explicitly. /cc @Iristyle, @glennsarti

**Comment by joshcooper** [comment]
Also are getting closer to being able to implement https://tickets.puppetlabs.com/browse/PUP-7586?

File: configs/projects/bolt-runtime.rb (hunk 1 above)
Comment (donoghuc) [reply configs/projects/bolt-runtime.rb]: Built bolt-runtime off latest commit. Runtime artifacts: builds.delivery.puppetlabs.net/puppet-runtime/444fed62ab170c6a3e7a7a9db077497971389a87/artifacts/

Using those runtimes I build bolt. Bolt packages with updated gems: http://builds.delivery.puppetlabs.net/puppet-bolt/963adf3253685416667f3d766588c84a3131184f/artifacts/

No problems with https://jenkins-master-prod-1.delivery.puppetlabs.net/view/bolt/job/platform_bolt-vanagon_vanagon-integration-suite_no-PEZ-daily-master/591/

So I'm good with the content from the bolt-runtime side.

**Comment by gimmyxd** [comment]
run into some issue running the agent pipleline using the new runtime with `ffi-win32-extensions`.
seems that what we `read_wide_string` from https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/api_types.rb#L56 is also defined in  `ffi-win32-extensions` https://github.com/chef/ffi-win32-extensions/blob/master/lib/ffi/win32/extensions.rb#L23 
The second one seems the overload the puppet defined `read_wide_string` which breaks functionality

cc: @joshcooper

**Comment by gimmyxd** [comment]
Closing this until further investigations are made

**Comment by glennsarti** [comment]
@joshcooper (sorry, late to party)

Yes. Although now that Chef have taken them over, a lot of fixes have been merged and being released more often.

**Pull request closed without merging**